src/MergeContext.js

```diff
@@ -454,6 +454,201 @@ class BranchPosition
     }
 }
 
+// Forward iterator for fields in the 'name:value' format.
+class FieldsTokenizer
+{
+    constructor(str) {
+        this._lines = str.split('\n');
+    }
+
+    // Extracts the next line from the input and checks that it matches the
+    // 'attrName:attrValue' pattern.
+    // Returns the matched {attrName, attrValue} pair.
+    nextField() {
+        assert(!this.atEnd());
+        const line = this._lines.shift();
+
+        if (/^\s/.test(line))
+            throw new Error(`the line starts with a delimiter: '${line}'`);
+
+        assert(line.trim().length > 0);
+
+        const pos = line.search(':');
+        if (pos < 0)
+            throw new Error(`':' is missing in the line: '${line}'`);
+
+        return {name: line.substring(0, pos), value: line.substring(pos+1)};
+    }
+
+    // whether we are at the end of input
+    atEnd() {
+        // treat an empty line as the end of input
+        return this._lines.length === 0 || this._lines[0].trim().length === 0;
+    }
+
+    // returns the unparsed (yet) string
+    remaining() {
+        return this._lines.join('\n');
+    }
+}
+
+// computes future commit message from raw PR
+class CommitMessage
+{
+    constructor(rawPr, defaultAuthor) {
+        // the (required) commit message title
+        this._title = rawPr.title + ' (#' + rawPr.number + ')';
+        // the main description part, without header and trailer
+        this._body = undefined;
+        // The optional finalizing description part with GitHub-related attributes,
+        // separated from the body by an empty line.
+        this._trailer = undefined;
+        // The 'Authored-by' meta information extracted from the optional description header,
+        // separated from the body by an empty line.
+        this._author = undefined;
+        // Author in the {name, email, date} format.
+        // The default author will be used in the future commit if the 'Authored-by' attribute is missing.
+        this._defaultAuthor = defaultAuthor;
+
+        this._checkRaw(this._title);
+
+        if (rawPr.body !== undefined && rawPr.body !== null) {
+            const trimmedPrDescription = this._trim(rawPr.body.replace(/\r+\n/g, '\n'));
+            this._checkRaw(trimmedPrDescription);
+            this._parse(trimmedPrDescription);
```

Comment:
I think each check/parse-like method should assume that it is given untrimmed input (and trim as needed). Otherwise, it is trivial to forget to trim "between" parts when splitting the raw description into parts and giving those parts to part-specific checkers/parsers...

```suggestion
            const prDescription = rawPr.body.replace(/\r+\n/g, '\n');
            this._checkRaw(prDescription);
            this._parse(prDescription);
```

.. and adjust the rest of the parsing code accordingly.